File: testprojects/scala_main/pipelines/PERF_SC_PIP/code/src/main/scala/org/main/scla_dep_mgmt_change/graph/PERF_AGGREGATE.scala
package org.main.scla_dep_mgmt_change.graph

import io.prophecy.libs._
import org.main.scla_dep_mgmt_change.config.ConfigStore._
import org.main.scla_dep_mgmt_change.config.Context
import org.main.scla_dep_mgmt_change.udfs.UDFs._
import org.main.scla_dep_mgmt_change.udfs._
import org.apache.spark._
import org.apache.spark.sql._
import org.apache.spark.sql.functions._
import org.apache.spark.sql.types._
import org.apache.spark.sql.expressions._
import java.time._

object PERF_AGGREGATE {

  def apply(context: Context, in: DataFrame): DataFrame =
    in.groupBy(col("cmls_3ds_authntn_mthd"))
      .agg(first(col("cmls_3ds_authntn_mthd")).as("cmls_3ds_authntn_mthd"),
           first(col("cmls_acct_cobrnd_bus_id_drvd"))
             .as("cmls_acct_cobrnd_bus_id_drvd")
      )

}

File: testprojects/python_main/pipelines/PERF_PY_PIP/code/com/main1/pythondepmanagement_1/graph/PERF_AGGREGATE.py
from pyspark.sql import *
from pyspark.sql.functions import *
from pyspark.sql.types import *
from prophecy.libs import typed_lit
from com.main1.pythondepmanagement_1.config.ConfigStore import *
from com.main1.pythondepmanagement_1.udfs.UDFs import *

def PERF_AGGREGATE(spark: SparkSession, in0: DataFrame) -> DataFrame:
    df1 = in0.groupBy(col("cmls_acct_ctry_cd_drvd"), col("cmls_acct_fundg_srce_cd_drvd"))

    return df1.agg(
        first(col("cmls_3ds_authntn_mthd")).alias("cmls_3ds_authntn_mthd"), 
        first(col("cmls_acct_cobrnd_bus_id_drvd")).alias("cmls_acct_cobrnd_bus_id_drvd")
    )
